codex/contracts/market.nim

```diff
@@ -36,7 +34,7 @@ proc approveFunds(market: OnChainMarket, amount: UInt256) {.async.} =
   let tokenAddress = await market.contract.token()
   let token = Erc20Token.new(tokenAddress, market.signer)
 
-  await token.approve(market.contract.address(), amount)
+  discard await token.approve(market.contract.address(), amount).confirm(1)
```

Comment:
`asyncSpawn` unless you really like suffering - `discard` will swallow exceptions.